selfdrive/debug/print_docs_diff.py

```diff
@@ -1,120 +1,101 @@
 
-FOOTNOTE_TAG = "<sup>{}</sup>"
-STAR_ICON = '<a href="##"><img valign="top" ' + \
-            'src="https://media.githubusercontent.com/media/commaai/openpilot/master/docs/assets/icon-star-{}.svg" width="22" /></a>'
-VIDEO_ICON = '<a href="{}" target="_blank">' + \
-             '<img height="18px" src="https://media.githubusercontent.com/media/commaai/openpilot/master/docs/assets/icon-youtube.svg"></img></a>'
-COLUMNS = "|" + "|".join([column.value for column in Column]) + "|"
-COLUMN_HEADER = "|---|---|---|{}|".format("|".join([":---:"] * (len(Column) - 3)))
-ARROW_SYMBOL = "➡️"
+def download_file(url, local_path):
+  response = requests.get(url)
+  response.raise_for_status()
+  with open(local_path, 'w') as f:
+    f.write(response.text)
 
 
-def load_base_car_docs(path):
-  with open(path, "rb") as f:
-    return pickle.load(f)
+def load_file(path):
+  with open(path, encoding='utf-8', errors='ignore') as file:
+    return file.read().splitlines()
 
 
-def match_cars(base_cars, new_cars):
-  changes = []
-  additions = []
-  for new in new_cars:
-    # Addition if no close matches or close match already used
-    # Change if close match and not already used
-    matches = difflib.get_close_matches(new.name, [b.name for b in base_cars], cutoff=0.)
-    if not len(matches) or matches[0] in [c[1].name for c in changes]:
-      additions.append(new)
-    else:
-      changes.append((new, next(car for car in base_cars if car.name == matches[0])))
+def parse_markdown(lines):
+  cars = {}
+  for line in lines:
+    if line.startswith('|') and not line.startswith('|---'):
+      columns = line.split('|')
+      platform = columns[1].strip()
+      model_year = columns[2].strip()
+      key = (platform, model_year)
+      cars[key] = line
+  return cars
 
-  # Removal if base car not in changes
-  removals = [b for b in base_cars if b.name not in [c[1].name for c in changes]]
-  return changes, additions, removals
 
+def extract_platform_model_year(line):
+  parts = line.split('|')
+  platform = parts[1].strip()
+  model_year = parts[2].strip()
+  return platform, model_year
 
-def build_column_diff(base_car, new_car):
-  row_builder = []
-  for column in Column:
-    base_column = base_car.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG)
-    new_column = new_car.get_column(column, STAR_ICON, VIDEO_ICON, FOOTNOTE_TAG)
 
-    if base_column != new_column:
-      row_builder.append(f"{base_column} {ARROW_SYMBOL} {new_column}")
-    else:
-      row_builder.append(new_column)
+def generate_diff(old_cars, new_cars):
+  changes = defaultdict(list)
+  all_keys = set(old_cars.keys()) | set(new_cars.keys())
+
+  for key in all_keys:
+    old_line = old_cars.get(key)
+    new_line = new_cars.get(key)
+
+    if old_line and new_line:
+      if old_line != new_line:
+        changes["column"].append(f"|{new_line}")
+    elif old_line:
+      platform, _ = key
+      for new_key in new_cars:
+        new_platform, new_model_year = new_key
```

Comment:
This is starting to look like the old version of this file, we should try to avoid parsing specific data as much as possible. Does this handle any column changing state without marking it as removed/added?